waku/waku_rendezvous/waku_peer_record.nim

```diff
@@ -0,0 +1,74 @@
+import std/times, sugar
+
+import
+  libp2p/[
+    protocols/rendezvous,
+    signed_envelope,
+    multicodec,
+    multiaddress,
+    protobuf/minprotobuf,
+    peerid,
+  ]
+
+type WakuPeerRecord* = object
+  # Considering only mix as of now, but we can keep extending this to include all capabilities part of Waku ENR
+  peerId*: PeerId
+  seqNo*: uint64
+  addresses*: seq[MultiAddress]
+  mixKey*: string
+
+proc payloadDomain*(T: typedesc[WakuPeerRecord]): string =
+  $multiCodec("libp2p-custom-peer-record")
+
+proc payloadType*(T: typedesc[WakuPeerRecord]): seq[byte] =
+  @[(byte) 0x05, (byte) 0x05]
+
+proc init*(
+    T: typedesc[WakuPeerRecord],
+    peerId: PeerId,
+    seqNo = getTime().toUnix().uint64,
+    addresses: seq[MultiAddress],
+    mixKey: string,
+): T =
+  WakuPeerRecord(peerId: peerId, seqNo: seqNo, addresses: addresses, mixKey: mixKey)
+
+proc decode*(
+    T: typedesc[WakuPeerRecord], buffer: seq[byte]
+): Result[WakuPeerRecord, ProtoError] =
+  let pb = initProtoBuffer(buffer)
+  var record = WakuPeerRecord()
+
+  ?pb.getRequiredField(1, record.peerId)
+  ?pb.getRequiredField(2, record.seqNo)
+  discard ?pb.getRepeatedField(3, record.addresses)
+
+  if record.addresses.len == 0:
+    return err(ProtoError.RequiredFieldMissing)
+
+  ?pb.getRequiredField(4, record.mixKey)
+
+  ok(record)
+
+proc encode*(record: WakuPeerRecord): seq[byte] =
```

Comment:
Encoding differs from what seems to be proposed in https://github.com/waku-org/specs/pull/95/files, specifically introduction of `seqNo` and no `protocols` field. Perhaps intentional?